code/datums/ai/dog/dog_controller.dm

```diff
@@ -0,0 +1,167 @@
+			blackboard[BB_DOG_FETCH_THROWER] = null
+			return
+		current_movement_target = return_target
+		current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/dog_deliver)
+		blackboard[BB_DOG_DELIVERING] = TRUE
+		return
+
+	var/list/old_throw_listeners = blackboard[BB_DOG_THROW_LISTENERS]
+	var/list/new_throw_listeners = list()
+
+	// check around us for people who we can hear throw things for fetching
+	// (would probably be smart to make this run less frequently than every tick)
+	for(var/i in range(AI_DOG_THROW_LISTEN_RANGE, get_turf(living_pawn)))
+		if(!iscarbon(i))
+			continue
+		var/mob/living/carbon/iter_carbon = i
+		if(!(iter_carbon in old_throw_listeners))
+			RegisterSignal(iter_carbon, COMSIG_MOB_THROW, .proc/listened_throw)
+		new_throw_listeners += iter_carbon
+		old_throw_listeners -= iter_carbon // we're still in, so remove them from the drop list
+
+	for(var/i in old_throw_listeners)
+		var/mob/living/carbon/lost_listener = i
+		UnregisterSignal(lost_listener, COMSIG_MOB_THROW)
+	blackboard[BB_DOG_THROW_LISTENERS] = new_throw_listeners
+
+/datum/ai_controller/dog/PerformIdleBehavior(delta_time)
+	// todo: port dogs eating snacks
+	var/mob/living/living_pawn = pawn
+
+	if(DT_PROB(25, delta_time) && (living_pawn.mobility_flags & MOBILITY_MOVE) && isturf(living_pawn.loc) && !living_pawn.pulledby)
+		var/move_dir = pick(GLOB.alldirs)
+		living_pawn.Move(get_step(living_pawn, move_dir), move_dir)
+
+/// Someone we were listening to throws for has thrown something, start listening to the thrown item so we can see if we want to fetch it when it lands
+/datum/ai_controller/dog/proc/listened_throw(mob/living/carbon/carbon_thrower)
+	if(blackboard[BB_DOG_FETCH_TARGET]) // we're already busy
+		return
+	var/obj/item/thrown_thing = carbon_thrower.get_active_held_item()
+	if(!isitem(thrown_thing) || get_dist(carbon_thrower, pawn) > AI_DOG_THROW_LISTEN_RANGE)
+		return
+	var/list/thrown_ignorelist = blackboard[BB_DOG_FETCH_TARGET_IGNORE]
+	if(thrown_thing in thrown_ignorelist)
+		return
+
+	RegisterSignal(thrown_thing, COMSIG_MOVABLE_THROW_LANDED, .proc/listen_throw_land)
+
+/// A throw we were listening to has finished, see if it's in range for us to try grabbing it
+/datum/ai_controller/dog/proc/listen_throw_land(obj/thrown_thing, datum/thrownthing/throwing_datum)
+	UnregisterSignal(thrown_thing, COMSIG_MOVABLE_THROW_LANDED)
+	if(!isitem(thrown_thing) || !isturf(thrown_thing.loc) || !(thrown_thing in view(pawn, AI_DOG_THROW_LISTEN_RANGE)))
+		UnregisterSignal(thrown_thing, list(COMSIG_PARENT_QDELETING, COMSIG_MOVABLE_THROW_LANDED))
+		return
+
+	current_movement_target = thrown_thing
+	blackboard[BB_DOG_FETCH_TARGET] = thrown_thing
+	blackboard[BB_DOG_FETCH_THROWER] = throwing_datum.thrower
+	current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/dog_fetch)
+
+/// Someone's interacting with us by hand, see if they're being nice or mean
+/datum/ai_controller/dog/proc/on_attack_hand(datum/source, mob/living/user)
```

Comment:
Are you removing the normal petting behavior as well?

Reply:
in simplemobs that is

Reply:
Nope, this is simply in addition to petting